ecosystem/python/sdk/aptos_sdk/async_client.py

```diff
@@ -527,6 +782,73 @@ async def transfer_object(
         )
         return await self.submit_bcs_transaction(signed_transaction)
 
+    async def view(
+        self,
+        function: str,
+        type_arguments: List[str],
+        arguments: List[str],
+        ledger_version: Optional[int] = None,
+    ) -> bytes:
+        """
+        Execute a view Move function with the given parameters and return its execution result.
+
+        The Aptos nodes prune account state history, via a configurable time window. If the requested ledger version
+        has been pruned, the server responds with a 410.
+
+        :param function: Entry function id is string representation of an entry function defined on-chain.
+        :param type_arguments: Type arguments of the function.
+        :param arguments: Arguments of the function.
+        :param ledger_version: Ledger version to get state of account. If not provided, it will be the latest version.
+        :returns: Execution result.
+        """
+        response = await self._post(
+            endpoint="view",
+            params={
+                "ledger_version": ledger_version,
+            },
+            headers={
+                "Accept": "application/json",
+                "Content-Type": "application/json",
+            },
+            data={
+                "function": function,
+                "type_arguments": type_arguments,
+                "arguments": arguments,
+            },
+        )
+        if response.status_code >= 400:
+            raise ApiError(response.text, response.status_code)
+
+        return response.content
+
+    async def _post(
+        self,
+        endpoint: str,
+        params: Optional[Dict[str, Any]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+        data: Optional[Dict[str, Any]] = None,
```

Comment:
wouldn't it be easier to just make this not optional and instead return a dict() by default instead?

beyond that, everything seems good

Reply:
With regard to making them not optional, I don't think so, as not every call would necessarily need to specify those (similar to the `.post` method in the underlying `httpx.AsyncClient`).

With regard to returning a `dict`, the function instead returns the actual `Response` object from the `httpx` library, such that any parent function using the method can understand the context around the response (i.e., the error code in the case of errors). 

Ideally, I would think that in future error handling and general "response processing" would be automatically handled in these utility methods (`_get()`, `_post()`, `_delete()`), but for now the method can just return the full response so that processing can be done in whatever parent method is calling the utility function.

Does that answer your question?

Reply:
I think the word optional is overloaded here...

more explicitly
```suggestion
        params: Dict[str, Any] = dict(),
```

as this eliminates line 832

similarly below on line 845

Reply:
That is a mutable default argument, which is said to be bad practice as it can cause very sneaky bugs

Reply:
Hmm... I thought `dict()` and `list()` wouldn't have the same issues, I stand corrected: https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
For a second I thought the same, but I imagine `dict()` would run and execute to `{}` before the function is actually defined (as opposed to only running when the function is called), so the effect would be the same as putting `{}` as the default argument.